src/Common/test/Common.Kubernetes.Test/IServiceCollectionExtensionsTest.cs

```diff
@@ -61,7 +63,6 @@ public void GetKubernetesApplicationOptions_ReturnsAndAddsOptions()
             // assert
             Assert.NotNull(options);
             Assert.Single(appInfos);
-            Assert.Equal(options, appInfos.FirstOrDefault());
```

Comment:
The exact same instance isn't returned because we're not building a temporary ServiceProvider in `AddKubernetesApplicationInstanceInfo` anymore